packages/integrations/markdoc/src/heading-ids.ts

```diff
@@ -64,7 +70,8 @@ export const heading: Schema = {
 };
 
 // Called internally to ensure `ctx` is generated per-file, instead of per-build.
-export function setupHeadingConfig(): HeadingIdConfig {
+export function setupHeadingConfig(experimentalHeadingIdCompat: boolean): HeadingIdConfig {
+	sluggerContext.experimentalHeadingIdCompat = experimentalHeadingIdCompat;
 	const headingSlugger = new Slugger();
 	return {
 		ctx: {
```

Comment:
Rather than mutating this global, could it be added to ctx?

Reply:
Good question. Found it really hard to reason about the Markdoc plugin which has a bunch of code specified in virtual module strings and then stuff like these context objects. I’ll take a look if that would work.